File: AMS_资讯信息组件/code/client/source/YssSecInformation.Support/Mp/SecMktMap/Service/IHlMktService.cs
////using YssBaseCls.Fun;
using FAST.Core.Util;
using FAST.Common.Service.Pojo;
using FAST.Common.Service.Services.Base;
using FAST.Core.Communication.Service;
using FAST.Platform.Workflow.Pojo;
using FAST.Common.Service.Pojo.Base;
using FAST.Core.Exceptions;


using FAST.Core.Context;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using FAST.Common.Service.Attributes;


////namespace YssData.Service.Mp
namespace YssSecInformation.Support.Mp.SecMktMap.Service
{
    /// <summary>
    /// 汇率行情数据服务接口
    /// author ll 20120925
    /// </summary>
    [NewServiceAttribute(ServiceType.NEW_SVC)]
    [ServiceAttribute(ServiceId = YssInformation.Support.common.ServiceIdConstant.OSGI_BASEBUSINESS, ServiceController = "com.yss.ams.sec.information.support.modules.mp.hlmkt.controller.IHlMktController", ServiceUrl = "")]
    public interface IHlMktService : IServiceBus
    {
        /// <summary>
        /// 条件查询
        /// </summary>
        /// <param name="quyCon">quyCon</param>
        /// <returns>查询结果对象</returns>
        [MethodAttribute(Method = RequestType.POST, MethodUrl = "/selectByCondition")]
        QueryRes selectByCondition(string quyCon);

        /// <summary>
        /// 根据组合、核算日期、交易币种加载货币和汇率信息 
        /// STORY #3705核算综合管理新增界面自动加载汇率 Added By xzl
        /// </summary>
        /// <param name="port">组合代码</param>
        /// <param name="accDate">核算日期</param>
        /// <param name="tradeCury">交易货币</param>
        /// <returns>汇率</returns>
        [MethodAttribute(Method = RequestType.POST, MethodUrl = "/queryErByCondition")]
        string queryErByCondition(string port, string accDate, string tradeCury);

    }
}

File: AMS_资讯信息组件/code/server/source/com.yss.ams.sec.information.support/src/main/java/com/yss/ams/sec/information/support/modules/mp/hlmkt/service/IHlMktService.java
package com.yss.ams.sec.information.support.modules.mp.hlmkt.service;

import com.yss.framework.api.mvc.biz.IServiceBus;
import com.yss.framework.api.restful.annotations.RestfulSupported;

@RestfulSupported
public interface IHlMktService extends IServiceBus {
    /////// By Jinghehe 2017-8-4 BUG #168158 资讯组件拆分出来，原来部分获取参数方法写的不规范 代码调整 
    /////// 挪到 IBaseValService 父类 接口当中 以便供其他界面调用 而不仅仅是凭证界面使用 核算管理数据 在YssData 项目中
    /////// IHlMktService 服务中的 queryErByCondition 方法废弃掉  
	/**
	 * 用于核算综合管理跟据组合，日期，和交易币种生成汇率表达式<br>
	 * Added By xzl 需求3705核算综合管理新增界面自动加载汇率
	 * @param port 组合
	 * @param accDate 核算日期
	 * @return
	 * @throws Exception
	 */
//	public String queryErByCondition(String port, String accDate, String tdCury)
//			throws Exception;
}
